File: auto-project/src/app/models/carro.ts
import { Marca } from "./marca";
import { Proprietario } from "./proprietario";

export class Carro {

  id!: number;
  nome!: string;
  ano!: number;
  cambio!: string;
  placa!: string;
  km!: string;
  combustivel!: string;
  financiamento!: string;
  marca!: Marca;
  proprietarios: Proprietario[] = [];

  constructor(
    nome: string,
    ano: number,
    cambio: string,
    placa: string,
    km: string,
    combustivel: string ,
    financiamento: string ,
    marca: Marca = { nome: '', cnpj: '' },
    proprietarios: Proprietario[] = [{ nome: '', idade: '' }]
  ) {
    this.nome = nome;
    this.ano = ano;
    this.cambio = cambio;
    this.placa = placa;
    this.km = km;
    this.combustivel = combustivel;
    this.financiamento = financiamento;
    this.marca = marca;
    this.proprietarios = proprietarios;
  }
}

File: auto-back-end-project/auto-app/auto-app/src/main/java/auto_app/auto_app/domain/Carro.java
package auto_app.auto_app.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
@Entity
public class Carro {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String nome;
    private int ano;

    @ManyToOne(cascade = CascadeType.PERSIST)
    @JsonIgnoreProperties("carros")
    private Marca marca;

    @ManyToMany(cascade = CascadeType.PERSIST)
    @JoinTable(name = "carro_proprietario")
    private List<Proprietario> proprietarios;
}
